webview/src/plots/components/App.test.tsx

```diff
@@ -1387,13 +1391,63 @@ describe('App', () => {
 
     expect(screen.queryByTestId('modal')).not.toBeInTheDocument()
 
-    const plot = within(screen.getAllByTestId(/^plot-/)[0]).getByRole('button')
+    const plot = within(screen.getAllByTestId(/^plot-/)[0]).getByTestId(
+      'zoomable-plot'
+    )
 
     fireEvent.click(plot)
 
     expect(screen.getByTestId('modal')).toBeInTheDocument()
   })
 
+  it('should open a modal with the plot zoomed in and actions menu open when clicking on a plot actions button', async () => {
+    renderAppWithOptionalData({
+      custom: customPlotsFixture
+    })
+
+    expect(screen.queryByTestId('modal')).not.toBeInTheDocument()
+
+    const plotActionsButton = within(
+      screen.getAllByTestId(/^plot-/)[0]
+    ).getByTestId('zoomable-plot-actions')
+
+    fireEvent.click(plotActionsButton)
+
+    expect(screen.getByTestId('modal')).toBeInTheDocument()
+    await waitFor(
+      () => {
+        expect(screen.getByTitle('Click to view actions')).toHaveAttribute(
+          'open'
+        )
+      },
+      { timeout: 1000 }
+    )
+  })
+
+  it('should open a modal with the plot zoomed in and actions menu open when key pressing on a plot actions button', async () => {
+    renderAppWithOptionalData({
+      custom: customPlotsFixture
+    })
+
+    expect(screen.queryByTestId('modal')).not.toBeInTheDocument()
+
+    const plotActionsButton = within(
+      screen.getAllByTestId(/^plot-/)[0]
+    ).getByTestId('zoomable-plot-actions')
```

Comment:
[C] Would have preferred `getByLabelText('See Plot Export Options')` here.

Reply:
https://derekndavis.com/posts/getbytestid-overused-react-testing-library

Reply:
Great points made in the article! I'll adjust the code to avoid using test ids.